tests/v2/validation/nodescaling/scale_replace_test.go

```diff
@@ -42,34 +45,58 @@ func (s *NodeScaleDownAndUp) SetupSuite() {
 	s.client = client
 }
 
-func (s *NodeScaleDownAndUp) TestEtcdScaleDownAndUp() {
-	s.Run("etcd-node-scale-down-and-up", func() {
-		ReplaceNodes(s.T(), s.client, s.client.RancherConfig.ClusterName, true, false, false)
-	})
-}
-
-func (s *NodeScaleDownAndUp) TestControlPlaneScaleDownAndUp() {
-	s.Run("controlplane-node-scale-down-and-up", func() {
-		ReplaceNodes(s.T(), s.client, s.client.RancherConfig.ClusterName, false, true, false)
-	})
-}
+func (s *NodeReplacingTestSuite) TestReplacingNodes() {
+	nodeRolesEtcd := machinepools.NodeRoles{
+		Etcd:         true,
+		ControlPlane: false,
+		Worker:       false,
+	}
+
+	nodeRolesControlPlane := machinepools.NodeRoles{
+		Etcd:         false,
+		ControlPlane: true,
+		Worker:       false,
+	}
+
+	nodeRolesWorker := machinepools.NodeRoles{
+		Etcd:         false,
+		ControlPlane: false,
+		Worker:       true,
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles machinepools.NodeRoles
+		client    *rancher.Client
+	}{
+		{"control plane nodes", nodeRolesControlPlane, s.client},
+		{"etcd nodes", nodeRolesEtcd, s.client},
+		{"worker nodes", nodeRolesWorker, s.client},
+	}
+
+	for _, tt := range tests {
+		clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
+		require.NoError(s.T(), err)
 
-func (s *NodeScaleDownAndUp) TestWorkerScaleDownAndUp() {
-	s.Run("worker-node-scale-down-and-up", func() {
-		ReplaceNodes(s.T(), s.client, s.client.RancherConfig.ClusterName, false, false, true)
-	})
-}
+		cluster, err := tt.client.Steve.SteveType(ProvisioningSteveResourceType).ByID(clusterID)
+		require.NoError(s.T(), err)
 
-func (s *NodeScaleDownAndUp) TestValidate() {
-	s.Run("rke2-validate", func() {
-		_, stevecluster, err := clusters.GetProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName, provisioninginput.Namespace)
+		updatedCluster := new(apisV1.Cluster)
+		err = v1.ConvertToK8sType(cluster, &updatedCluster)
 		require.NoError(s.T(), err)
 
-		clusterConfig := clusters.ConvertConfigToClusterConfig(s.clustersConfig)
-		provisioning.VerifyCluster(s.T(), s.client, clusterConfig, stevecluster)
-	})
+		if strings.Contains(updatedCluster.Spec.KubernetesVersion, "rke2") {
+			tt.name = "Replacing RKE2 " + tt.name
+		} else {
+			tt.name = "Replacing K3S " + tt.name
+		}
+
```

Comment:
couldn't you use similar logic for rke1 and remove the rke1 test file?

Reply:
That seems a bit excessive IMO, considering we already have a separate rke1 test file. I don't see why we wouldn't just keep it as it is.

Reply:
I see what you're saying. Considering shepherd PRs have already been merged, it's best practice to not make edits to the rancher/rancher PRs outside of just rebasing to update the shepherd version.

Let's look at that in a separate PR.

Reply:
but this way, I have to run a completely different test if I want to run rke1. If its all in one file, its usually simpler for the tester. plus, less code/files to maintain

Reply:
^ discussed further offline. Will do this in a separate PR shortly after this PR gets merged.